java/adapter/avro/src/main/java/org/apache/arrow/consumers/AvroBytesConsumer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.consumers;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import org.apache.arrow.vector.VarBinaryVector;
+import org.apache.arrow.vector.complex.impl.VarBinaryWriterImpl;
+import org.apache.arrow.vector.complex.writer.VarBinaryWriter;
+import org.apache.arrow.vector.holders.VarBinaryHolder;
+import org.apache.avro.io.Decoder;
+
+/**
+ * Consumer which consume bytes type values from avro decoder.
+ * Write the data to {@link VarBinaryVector}.
+ */
+public class AvroBytesConsumer implements Consumer {
+
+  private final VarBinaryWriter writer;
+  private final VarBinaryVector vector;
+
+  public AvroBytesConsumer(VarBinaryVector vector) {
+    this.vector = vector;
+    this.writer = new VarBinaryWriterImpl(vector);
+  }
+
+  @Override
+  public void consume(Decoder decoder) throws IOException {
+    VarBinaryHolder holder = new VarBinaryHolder();
+    ByteBuffer byteBuffer = decoder.readBytes(null);
```

Comment:
we should probably keep a cached byte buffer for values below a certain size.  

Even better would be if we could create a ByteBuffer that wraps the Vector and puts bytes directly where they need to go.  This can be done in a separate PR.

Reply:
Can we address the first part of my comment (caching a ByteBuffer instead of creating a new one each time)?

Otherwise LGTM as start.

Reply:
Sure, I will create a new JIRA to address this, thanks!